### PR TITLE
fix missing t2 strs

### PR DESCRIPTION
**Pull request by Jsims0**
- fixed the incorrect conversion of STR tier labels 
- added testing for STR tier labeling

### CODE REVIEW

**Review by RSWilson1 — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22)* status: 0 of 3 files reviewed, 5 unresolved discussions

___
*[`resources/home/dnanexus/get_variant_info.py` line 212 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsNrmELU4Im6hR4sP:-ONPsNrmELU4Im6hR4sQ:b9xh5vh) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/get_variant_info.py#L212)):*
> ```Python
>         variant (dict): dict extracted from JSON describing single variant
>         proband (str): GEL ID for the proband
>         columns (list): list of columns to make into keys for variant dict
> ```

could you add ev_idx here
___
*[`resources/home/dnanexus/make_workbook.py` line 635 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsSBdCnJuyOcjof_Q:-ONPsSBdCnJuyOcjof_R:b-zamcj4) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/make_workbook.py#L635)):*
> ```Python
>                 if event["tier"] in ["TIER1", "TIER2"]:
>                     event_index = s_t_r["reportEvents"].index(event)
>                     print(f"event tier: {event['tier']}:event index: {event_index}")
> ```

Is this print needed/useful? Or leftover from testing?
___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 175 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsF6iEMBAIB9oft49:-ONPsF6iEMBAIB9oft4A:buc168l) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L175)):*
> ```Python
>         ) == {"ColA": '', "ColB": ''}
> 
>     def test_get_str_info_teir1(self):
> ```

typo "tier"
___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 175 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsrbf38GitBKVUVgh:-ONPsrbf38GitBKVUVgi:bnsd5cp) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L175)):*
> ```Python
>         ) == {"ColA": '', "ColB": ''}
> 
>     def test_get_str_info_teir1(self):
> ```

Also this seems to state it is testing tier 1 STRs but tests both tiers? Could you add tests to cover each? Also a test to cover presence of tier=null would be good.
___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 242 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPslvv54Ya3V3hMgD2:-ONPslvv54Ya3V3hMgD3:bdk7ku9) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L242)):*
> ```Python
>         # Call the function
>         result = var_info.get_str_info(variant, proband, columns, ev_idx)
>         # Modify the variant to replace reportEvents with TIER2 events
> ```

Why is this needed?

**Review by Jsims0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22)* status: 0 of 3 files reviewed, 4 unresolved discussions (waiting on @RSWilson1)

___
*[`resources/home/dnanexus/get_variant_info.py` line 212 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsNrmELU4Im6hR4sP:-ONQBUA25dVQDYCmyZkx:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/get_variant_info.py#L212)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

could you add ev_idx here
</blockquote></details>

Done.
___
*[`resources/home/dnanexus/make_workbook.py` line 635 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsSBdCnJuyOcjof_Q:-ONQ8mjRBgo1sf-i8dzK:b-gh38lo) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/make_workbook.py#L635)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Is this print needed/useful? Or leftover from testing?
</blockquote></details>

Nope, will remove, thought I did but must have saved after I added the deletion to the commit
Done.
___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 175 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPsrbf38GitBKVUVgh:-ONUJzJt6T2JUnsyiLmQ:bfahr70) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L175)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Also this seems to state it is testing tier 1 STRs but tests both tiers? Could you add tests to cover each? Also a test to cover presence of tier=null would be good.
</blockquote></details>

Done. the filtering of which STRs to report is done outside of get_str_info 
___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 242 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPslvv54Ya3V3hMgD2:-ONQCYFKElbGY2U9gZXf:b-9yn1gn) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L242)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Why is this needed?
</blockquote></details>

changes the mock variant to only have tier 2 STRs so it should be reported 'TEIR2_STR' in output, more convenient than having multiple mock JSONs written out.

File: resources/home/dnanexus/get_variant_info.py [hunk 1]
```diff
@@ -226,7 +227,9 @@ def get_str_info(variant, proband, columns):
     var_dict["End"] = variant["coordinates"]["end"]
     var_dict["Length"] = abs(var_dict["End"] - var_dict["Pos"])
     var_dict["Type"] = "STR"
-    var_dict["Priority"] = "TIER1_STR"
+    var_dict["Priority"] = convert_tier(
+        variant["reportEvents"][ev_idx]["tier"], "STR")
+
```
[the remark this answers is not in the document]
Comment (RSWilson1) [reply resources/home/dnanexus/get_variant_info.py:232]: I don't think this is needed and would be best for it to fail rather than continuing with unknown_tier.

**Review by RSWilson1 — APPROVED** [review]
> Reviewed 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> Dismissed @coderabbitai[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Jsims0)

___
*[`resources/home/dnanexus/tests/test_make_workbook.py` line 242 at r1](https://reviewable.io/reviews/eastgenomics/eggd_generate_rd_wgs_workbook/22#-ONPslvv54Ya3V3hMgD2:-ONUd4ZK4KyIC6O69paC:b-2jaqoz) ([raw file](https://github.com/eastgenomics/eggd_generate_rd_wgs_workbook/blob/e1742bde430f9659c5940679672f1afffecf8d1c/resources/home/dnanexus/tests/test_make_workbook.py#L242)):*
<details><summary><i>Previously, Jsims0 wrote…</i></summary><blockquote>

changes the mock variant to only have tier 2 STRs so it should be reported 'TEIR2_STR' in output, more convenient than having multiple mock JSONs written out.
</blockquote></details>

Nice, testing looks more comprehensive now.
